### PR TITLE
Adds logging information about puma threads.

### PR DESCRIPTION
**Pull request by amprokop**
Adds logging information about Puma threads into the app. This **should** emit a log statement every time the `ThreadPool` class is used.

^^ v. true

Testing: observed this statement in the logs. 
`Puma stats -- Process pid: 27156 Total threads: 1 Backlog of actions: 0 Waiting threads: 1 Active threads: 0Live threads: 1/1 alive`

### CODE REVIEW

File: config.ru [hunk 1]
```diff
@@ -23,4 +23,53 @@ use Prometheus::Client::Rack::Collector
...
+        end
+      end
+    end
+    super *args
+  end
+end
+
+module Puma
+  class ThreadPool
+    prepend PumaThreadLogger
```
Comment (askldjd) [inline config.ru]: Very cool, never knew you could do this.

File: config.ru [hunk 2]
```diff
@@ -23,4 +23,53 @@ use Prometheus::Client::Rack::Collector
...
 use Prometheus::Client::Rack::Exporter
 
+
+# TODO (alex): this should be a temporary addition to try to solve
+# a deployment bug. We should refactor or remove this after it serves its purpose.
+module PumaThreadLogger
+  def initialize *args
+    Thread.new do
+      loop do
+        sleep 1
```
Comment (askldjd) [inline config.ru]: This will flood the log files a bit. Would this be better if we can toggle this with a env variable?

File: config.ru (hunk 1 above)
Comment (amprokop) [inline config.ru]: yeah :) it's dangerous for sure. probably bad style.

**Comment by amprokop** [comment]
@joofsh @askldjd — actually, i want to fix this up a bit. I found something weird here because of the following logs
```
Puma stats -- Process pid: 29202 Total threads: 6 Backlog of actions: 0 Waiting threads: 6 Active threads: 0 Live threads: 6/6 alive
Puma stats -- Process pid: 29202 Total threads: 6 Backlog of actions: 0 Waiting threads: 6 Active threads: 0 Live threads: 6/6 alive
Puma stats -- Process pid: 29202 Total threads: 5 Backlog of actions: 0 Waiting threads: 5 Active threads: 0 Live threads: 5/5 alive
Puma stats -- Process pid: 29202 Total threads: 5 Backlog of actions: 0 Waiting threads: 5 Active threads: 0 Live threads: 5/5 alive
Puma stats -- Process pid: 29202 Total threads: 4 Backlog of actions: 0 Waiting threads: 4 Active threads: 0 Live threads: 4/4 alive
Puma stats -- Process pid: 29202 Total threads: 4 Backlog of actions: 0 Waiting threads: 4 Active threads: 0 Live threads: 4/4 alive
Puma stats -- Process pid: 29202 Total threads: 3 Backlog of actions: 0 Waiting threads: 3 Active threads: 0 Live threads: 3/3 alive
Puma stats -- Process pid: 29202 Total threads: 3 Backlog of actions: 0 Waiting threads: 3 Active threads: 0 Live threads: 3/3 alive
Puma stats -- Process pid: 29202 Total threads: 1 Backlog of actions: 0 Waiting threads: 1 Active threads: 0 Live threads: 1/1 alive
Puma stats -- Process pid: 29202 Total threads: 1 Backlog of actions: 0 Waiting threads: 1 Active threads: 0 Live threads: 1/1 alive
Puma stats -- Process pid: 29202 Total threads: 1 Backlog of actions: 0 Waiting threads: 1 Active threads: 0 Live threads: 1/1 alive
Puma stats -- Process pid: 29202 Total threads: 1 Backlog of actions: 0 Waiting threads: 1 Active threads: 0 Live threads: 1/1 alive
```

**Comment by amprokop** [comment]
i didn't fix this up a bit. i figure the ROI on trying to debug this now is higher than making it entirely correct :) warning to all who see this: it doesn't follow our style guide. we have a simmering-just-below emergency bug though, and this hack might help to try to get more information about it. do not fear, it will be gone soon.

File: config.ru (hunk 2 above)
Comment (amprokop) [inline config.ru]: done

**Comment by aroltsch** [comment]
👍

**Pull request merged**